src/ua.c

```diff
@@ -1982,6 +2021,59 @@ struct sipevent_sock *uag_sipevent_sock(void)
 }
 
 
+static bool uri_match_transport(const struct uri *accu, enum sip_transp tp)
+{
+	struct pl pl;
+	int err;
+
+	err = msg_param_decode(&accu->params, "transport", &pl);
+	if (err)
+		return true;
+
+	return !pl_strcasecmp(&pl, sip_transp_name(tp));
+}
+
+
+static bool uri_match_af(const struct uri *accu, const struct uri *peeru)
+{
+#ifdef HAVE_INET6
+	struct sa sa1;
+	struct sa sa2;
+	int err;
+#endif
+
```

Comment:
the uri has an `int af` parameter, that has the following meaning:

- AF_UNSPEC domain
- AF_INET IPv4
- AF_INET6 IPv6

if uri is a domain (i.e. af=AF_UNSPEC) I believe that this check should be skipped.

Reply:
I move these lines up to the top of the function.

```
	/* if peer uri is a domain, we simply return true */
	if (peeru->af == AF_UNSPEC)
		return true;
```